### DISCUSSION TITLE
意见反馈区(旧)

### DISCUSSION THREAD

**Discussion by VermiIIi0n** [VermiIIi0n/fuckZHS#20] [General]
### 如果有什么想法但是懒得开 issue 的话, 欢迎在此处留言🌴

* 此处仅是想法闲聊, 请勿在此反馈 BUG, 有专门模版用于反馈 BUG

**Comment by Sn0r1axi** [comment]
Fucking Zhidao course: 大学生安全文化
  |
  |__Fucking chapter 安全观念概述篇
  |  |
  |  |__Fucking lesson 安全价值和安全文化的意义
  |  |  |__Fucking video 数字说安全
  |  |  |__Fucking video 视频看灾难
  |  |  |##Failed: code not found in {'timestamp': 1660709829185, 'status': 404, 'error': 'Not Found', 'message': 'Not
  |  |  |__Fucking video 课程的内容
  |  |  |##Failed: code not found in {'timestamp': 1660709847524, 'status': 404, 'error': 'Not Found', 'message': 'Not
  |  |
  |  |__Fucking lesson 中国古代安全文化及其启示
  |  |  |__Fucking video 安全民俗文化的有趣故事
  |  |  |##Failed: code not found in {'timestamp': 1660709865811, 'status': 404, 'error': 'Not Found', 'message': 'Not
  |  |  |__Fucking video 四灵兽传说及古代安全技术点滴
  |  |  |##Failed: code not found in {'timestamp': 1660709884149, 'status': 404, 'error': 'Not Found', 'message': 'Not
  |  |  |__Fucking video 中国古代安全思想方略
  |  |  |##Failed: code not found in {'timestamp': 1660709902453, 'status': 404, 'error': 'Not Found', 'message': 'Not

**Comment by VermiIIi0n** [comment]
~信息不足, 麻烦按模版开个 BUG Issue~

_*Edit:_ 已解决

**Comment by TanYu-G** [comment]
哥 为啥每次扫码都显示二维码过期

**Comment by VermiIIi0n** [comment]
> 哥 为啥每次扫码都显示二维码过期

我刚测试了没问题啊, 先换个网络或者设备试试, 还有问题的话麻烦按模版开个 Issue

**Comment by YANGNIGHT** [comment]
![图片](https://user-images.githubusercontent.com/79457406/189255738-6745e3cd-4d14-4276-9228-8eadcb1edde6.png)
我这是模块导入失败吗

**Comment by VermiIIi0n** [comment]
刚用错号了, 两次回复打扰了

Windows 的 bug, 但我这边 Win 10/11 测试是正常的, 不知道你那为啥报错.

先把utils.py下的length = (os.get_terminal_size ().columns-4), fill = '#'): 里的第一个参数改成80试试吧

不过建议用 Windows 的 Linux 子系统跑, 我很少在 Win 上跑.

**Comment by YANGNIGHT** [comment]
放弃win了 linux可以 不过我想每次只刷三十分钟 能直接在源码里改吗 初学者还不晓得在哪里改

**Comment by VermiIIi0n** [comment]
可以使用命令行参数 `-l` 限制时长, 更多参数可以看[文档](https://github.com/VermiIIi0n/fuckZHS#%E5%91%BD%E4%BB%A4%E8%A1%8C%E5%8F%82%E6%95%B0%E5%88%97%E8%A1%A8)

比如`python ./main.py -l 30` 就可以限制30分钟了

**Comment by YANGNIGHT** [comment]
> 可以使用命令行参数 `-l` 限制时长, 更多参数可以看文档
> 
> 比如`python ./main.py -l 30` 就可以限制30分钟了

好的 这个30分钟是程序提交的课程时长 还是实际运行时长 还是这两个是一样的提交时长就是实际运行时长

**Comment by VermiIIi0n** [comment]
> > 可以使用命令行参数 `-l` 限制时长, 更多参数可以看文档
> > 比如`python ./main.py -l 30` 就可以限制30分钟了
> 
> 好的 这个30分钟是程序提交的课程时长 还是实际运行时长 还是这两个是一样的提交时长就是实际运行时长

是实际运行时长, 提交时长是运行时长*播放速度, 学分课播放速度默认就是 1.5, 也可以用参数改

**Comment by p1p1Shr1mp** [comment]
大佬，我用参数-l无法限制时长，我设1测试发现会一直跑

**Comment by VermiIIi0n** [comment]
放完当前视频就不会继续下一个了, 当时是为了更符合正常观看习惯这样写的, 时间太久自己都忘了🌚

**Comment by p1p1Shr1mp** [comment]
每天早上起来都能接到知到的异常检测警告但是没有异常学习记录...

**Comment by VermiIIi0n** [comment]
> 每天早上起来都能接到知到的异常检测警告但是没有异常学习记录...

什么警告呀? 我目前还没见过...

**Comment by p1p1Shr1mp** [comment]
> > 每天早上起来都能接到知到的异常检测警告但是没有异常学习记录...
> 
> 什么警告呀? 我目前还没见过...

就是提示我检测到学习异常让我诚信学习...但是app里看不到异常学习记录也挺奇怪的

**Comment by VermiIIi0n** [comment]
> > > 每天早上起来都能接到知到的异常检测警告但是没有异常学习记录...
> > 
> > 
> > 什么警告呀? 我目前还没见过...
> 
> 就是提示我检测到学习异常让我诚信学习...但是app里看不到异常学习记录也挺奇怪的

你是用默认的参数吗? 按理说和浏览器观看差异很小. 它那个主要是检测脚本注入和异常播放速度来判断的.
如果愿意的话也可以开个 issue 传传日志让我研究下.

**Comment by 1660962168** [comment]
请问一下二维码登录 linux系统图片保存在什么位置啊 想挂服务器上跑

**Comment by VermiIIi0n** [comment]
> 请问一下二维码登录 linux系统图片保存在什么位置啊 想挂服务器上跑

~放在内存~  是直接用 `PIL` 显示的, 如果要用服务器得打印二维码到终端, 使用 `--show_in_terminal` 就行

不过目前版本的二维码很大...得把终端尺寸搞大点才能完全显示.

**Comment by LFMJUN** [comment]
在终端上面输出二维码扫描以后，出现了
![b664273db9a42603510f6706b185c2d](https://user-images.githubusercontent.com/47923014/190988674-b13b5243-a4bc-41cb-8e91-02864337954f.jpg)
没有任何反应，尝试访问电脑版然后登录也没有反应，开了debug也没有看到错误

**Comment by VermiIIi0n** [comment]
> 在终端上面输出二维码扫描以后，出现了  没有任何反应，尝试访问电脑版然后登录也没有反应，开了debug也没有看到错误

那就下载呀, 得用知到客户端扫

**Comment by YoumyTheGoust** [comment]
哥，二维码有点太长了扫不出来

**Comment by YoumyTheGoust** [comment]
用知到app扫的话总是扫不出来，检测不到二维码

**Comment by VermiIIi0n** [comment]
> 用知到app扫的话总是扫不出来，检测不到二维码

目前的打印到终端的二维码确实大...下版本改进。
但是PIL显示出来的应该没问题。

最近太忙了，只能过几天再更新了

**Comment by YoumyTheGoust** [comment]
> > 用知道到app扫描的话总是扫不出来，检测不到二维码
> 
> 目前的但到终端的二维码并没有大...打印下 显示改进。
> 
> 最近太忙了，只能过几天再更新了

好哒，辛苦大佬 了

**Comment by VermiIIi0n** [comment]
> > > 用知道到app扫描的话总是扫不出来，检测不到二维码
> > 
> > 
> > 目前的但到终端的二维码并没有大...打印下 显示改进。
> > 最近太忙了，只能过几天再更新了
> 
> 好哒，辛苦大佬 了

姑且弄了个小很多的二维码, 但依赖于字体, 可能打印出来会很丑. 
先运行次 _main.py_ 以更新配置文件, 然后把 _config.json_ 里的 `ensure_unicode` 改成 `true` 就行.

**Comment by ChunQiu2076** [comment]
问题1：如果不修改config.json里的ensure_unicode（默认为false）控制台输出二维码颜色正确，但只会输出一半
问题2：如果修改config.json里的ensure_unicode（改为true）控制台输出二维码正常，但是颜色为反色（扫描不识别）可以通过ps进行反色
测试环境：win10、vscode、（使用pycharm会有奇奇怪怪的异常）
建议1：可不可以加入互动分
建议2：可不可以加入见面课
建议3：可不可以打包成exe可执行文件，或者可视化界面，选择参数后执行。例如：输入1=学习25分钟。输入2=学习见面课......

**Comment by VermiIIi0n** [comment]
> 问题1：如果不修改config.json里的ensure_unicode（默认为false）控制台输出二维码颜色正确，但只会输出一半
> 问题2：如果修改config.json里的ensure_unicode（改为true）控制台输出二维码正常，但是颜色为反色（扫描不识别）可以通过ps进行反色
> 测试环境：win10、vscode、（使用pycharm会有奇奇怪怪的异常）
> 建议1：可不可以加入互动分
> 建议2：可不可以加入见面课
> 建议3：可不可以打包成exe可执行文件，或者可视化界面，选择参数后执行。例如：输入1=学习25分钟。输入2=学习见面课......

1. 只会输出一半可能和终端尺寸有关，没在 PowerShell 上测试过，尺寸拉大点应该能完全显示的。如果还是有问题可以把 `show_in_terminal` 关掉，这样就回退到以前那样直接显示图片。

2. Unicode 模式下样式和终端字体主题等有关，只是因为小一些而且能在非 TTY 终端下使用而加入的，一般不推荐用。把终端字体改黑色，背景浅色就能用。（直接换 iOS 端知到也是个办法）

3. 互动分与见面课因为我没有这些内容也不太好测试，而且 [luoyily](https://github.com/luoyily/zhihuishu-tool) 的脚本支持这些功能，所以就暂时先鸽了。

4. GUI 的话，最早是为了全自动执行而去的就没写，后来因为登录验证被迫变成每次都要扫二维码。目前还是打算先想办法绕过登录验证，这样就不需要交互了。不过加个 GUI 也没啥毛病，得看学校什么时候让我闲一下吧...

总之谢谢反馈～

**Comment by Biscuit19** [comment]
大佬太牛逼了！这脚本跑的真帅。
分享一个小问题，win10环境，用pycharm跑程序会报这个句柄错误
![image](https://user-images.githubusercontent.com/92147158/194498570-f4888219-4cd6-4494-9900-a938ad605efc.png)
但如果用vscode跑就能正常运行，前面用windows跑出这个问题的兄弟可以用vscode试一下。
最后再次感谢大佬，太帅了这脚本

**Comment by VermiIIi0n** [comment]
当初没想到会有这么多人反馈, 非常感谢各位的参与, 让我们能更好的 fuck 智慧树(

现在这个 issue 下留言有点长了, 可能不太方便用来当讨论区了 ~(而且我也见不得有 issues 🌚)~ , 故开启了 Discussions, 之后在 [这里](https://github.com/VermiIIi0n/fuckZHS/discussions/19) 留言或者新开讨论区都行.

再次感谢各位参与交流~
